### ISSUE TITLE
CIDER 2.0: The Plan

### ISSUE THREAD

**Issue by bbatsov** [clojure-emacs/cider#3356]
I'm thinking of cutting one final CIDER 1.x release soon, before #3352 is finished, and then prepare for a CIDER 2.0 release. Why so? 

- the logjam library targets Clojure 1.9, so it feels like a good time to drop Clojure 1.8 support, given that almost no one uses it
  - we'll also be able to simplify the code related to clojure.spec
- a new major version will allows to tweak some defaults, like the ones we've recently discussed with @yuhan0 
- might be a good opportunity to make bigger changes to the connection management

Ideas/concerns are welcome!

**Comment by yuhan0** [comment]
Strongly agree about `cljr-slash`, in fact it's practically the only feature that I use at all from clj-refactor, after all its "dumber" refactorings were migrated over to clojure-mode.

Another nice thing I'd like to see merged into Cider would be the `cljr-add-ns-to-blank-clj-files` functionality and a simpler file-renaming functionality - no global AST parsing across the entire project needed, just compare the actual ns name against the inferred one and offer to fix it up in the current buffer.

Most of cljr's other features feel a bit too idiosyncratic and dependency-heavy to me - eg. pulling in the entirety of yasnippet, hydra and multiple-cursors, and I don't think it would be a good idea to merge those parts into Cider as-is.

**Comment by jmckitrick** [comment]
I've always had issues getting clj-refactor to work correctly. I'm sure I could push through it, but never had the patience. I'd love to see it work 'out of the box'.

**Comment by rrudakov** [comment]
Maybe it would be nice to make CIDER work with `clojure-ts-mode`?

**Comment by bbatsov** [comment]
@rrudakov I think it's a bit too early for this, given that `clojure-ts-mode` requires some more work before we can consider it feature-complete.

**Comment by rrudakov** [comment]
> @rrudakov I think it's a bit too early for this, given that `clojure-ts-mode` requires some more work before we can consider it feature-complete.

I think it could help to bring more contributors to improve tree sitter support for Clojure. Now it's problematic to use clojure-ts-mode for daily work, because of lack of CIDER support.

**Comment by alexander-yakushev** [comment]
FWIW, if you plan to drop 1.8, it makes sense to move straight to 1.10 as the minimum. 1.10 contains several important fixes related to related to Java9+ compatibility[1], so I doubt many people who use Java9+ would stay on 1.9. Besides, nothing controversial was introduced between 1.9 and 1.10 that would prevent people from upgrading.

[1] https://github.com/clojure/clojure/blob/master/changes.md#1-compatibility-and-dependencies

**Comment by bbatsov** [comment]
@alexander-yakushev Yeah, I was thinking the same thing - less than 5% of State of Clojure 2023 respondents are using Clojure 1.9 anyways. And there's nothing preventing them to upgrade, given that CIDER has required JDK 8+ for a very long time. I think we'll target Clojure 1.10 directly.

**Comment by bbatsov** [comment]
Funny enough we've done many of things discussed above without getting to CIDER 2.0. 😆 Oh, well...

I plan to think more about the roadmap for it after we have some final State of CIDER 2024 survey results. (see #3765)
